### DISCUSSION TITLE
restrict access by client's ip address

### DISCUSSION THREAD

**Discussion by bolerap** [vercel/next.js#12588] [Help]
hi all.
How can i get client's ip address. for now, this is my code in middleware to check this 
`
    const ip = req.headers["x-real-ip"] || req.headers["x-forwarded-for"] || req.connection.remoteAddress;
    console.log("ip: ", ip);
    console.log("req headers: ", req.headers);
`
and this is log from my server (not local)
`
  ip:  ::ffff:127.0.0.1
  req headers:  {
    host: '127.0.0.1:8891',
    connection: 'close',
    'cache-control': 'max-age=0',
    'upgrade-insecure-requests': '1',
    'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4136.1 Safari/537.36',
    accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
    'sec-fetch-site': 'none',
    'sec-fetch-mode': 'navigate',
    'sec-fetch-user': '?1',
    'sec-fetch-dest': 'document',
    'accept-encoding': 'gzip, deflate, br',
    'accept-language': 'en-US,en;q=0.9'
  }
 `

How can i get ip address from visitor. many thanks

**Comment by huykon** [comment]
Do you guys know any way to do this?
